packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -692,9 +692,11 @@ export class SnapController extends BaseController<
         snapStates: {
           persist: () => {
             return Object.keys(this.state.snaps).reduce<
-              Record<string, string | null>
+              Record<ValidatedSnapId, SnapRuntimeData['state']>
             >((acc, cur) => {
-              acc[cur] = this.#getRuntimeExpect(cur).state;
+              acc[cur as ValidatedSnapId] = this.#getRuntimeExpect(
```

Comment:
Do we still need the cast when we use the generic type in `reduce`? 🤔

Reply:
Yes, it complains that you can't use string to index `Record<ValidatedSnapId, SnapRuntimeData['state']>`